### ISSUE TITLE
Project setup & naming conventions

### ISSUE THREAD

**Issue by 11janci** [GSG-G7/Parent-assistent#68]
Hi all,

I'm Jan, I will be helping you with the project as a mentor. Any questions, pls don't hesitate to get in touch (let me know what is your preferred communication channel)

I have some minor initial comments on the project:

**Use PascalCase filenames only for components, not generic directories and other js files**

Some generic guidelines:
- a js file containing a component should be PascalCase (for example "Button.js")
- a directory containing an index.js with a component should be PascalCase (for example "Button")
  - so that the component can be imported in the code like this: `import Button from './components/Button';` 
- everything else should be kebab-case or camelCase (decide which one you like more as a team and stick to it): pls rename the directories `Components`, `Pages` and `CommonComponent` to `components`, `pages` and `common-component` (more on the last one see below)

It might seem like insignificant details, but it does make a difference - seeing a PascalCase directory gives me a hint that it contains a component; it is misleading when it actually doesn't..

**Other tips for naming**
- use plurals for directory names holding multiple similar but unrelated modules - `common-components` instead of `common-component`, `utils` or `helpers` for helper functions, `controllers`, etc.
- do not repeat words from higher modules/directories in the lower ones, or use generic words in the names. Instead of having `components/common-components` you can have `components/common` (or `components/shared`).
- [naming things](https://www.mediawiki.org/wiki/Naming_things) is hard, but good names make a lot of difference!

**Use propTypes and defaultProps**
- helps prevent and detect passing wrong properties to a component
- https://reactjs.org/docs/typechecking-with-proptypes.html
- https://www.npmjs.com/package/prop-types

------------
To sum up, here are the AC:
- rename:
  - `Components` -> `components`
  - `Pages` -> `pages`
  - `CommonComponent` -> `common` or `shared` (other suggestions welcome)
- introduce propTypes and defaultProps into the project (at least one component using it) and start enforcing it in PRs

Overal, good start guys! 👍

**Comment by aminalakhsham** [comment]
hi, Jan
I amin the team leader to this project 
thank you for helping us and this helpful notes and review
we will so happy with you and with helpful notes and review 
about communication channel:
we have a server in discord app if you want to join with us (just tell me)

**Comment by Asmaathabet** [comment]
@11janci  
Hi Jan , 
I'm Asmaa , Thanks for your suggestions and I will edit all the names as you said ^_^ .

**Comment by AlaaTaima** [comment]
Hi Jan ,
Iam Alaa 
thanks for your useful details , it's a good way of naming folder,files and components. 
I will edit names as you explains.

**Comment by 11janci** [comment]
> @11janci
> hi, Jan
> I'm Amin the team leader to this project
> thank you for helping us and this helpful notes and review
> we will so happy with you and with helpful notes and review
> about communication channel:
> we have a server in discord app if you want to join with us (just tell me)

Hi Amin,
thanks; yes you can add me to your discord. My handle is the same as here on github (or you can send me your server's details).

**Comment by ali-7** [comment]
@11janci thank you Jan, I'll add you to the discord channels for the teams.
